crypto/src/main/kotlin/web5/sdk/crypto/jwk/Jwk.kt

```diff
@@ -0,0 +1,180 @@
+package web5.sdk.crypto.jwk
+
+import web5.sdk.common.Convert
+import web5.sdk.common.Json
+import web5.sdk.crypto.Ed25519
+import java.security.MessageDigest
+
+/**
+ * Represents a [JSON Web Key (Jwk )](https://datatracker.ietf.org/doc/html/rfc7517).
+ * A Jwk is a JSON object that represents a cryptographic key. This class
+ * provides functionalities to manage a Jwk including its creation, conversion
+ * to and from JSON, and computing a thumbprint.
+ *
+ * Example:
+ * ```
+ * var jwk = Jwk(
+ *   kty: 'RSA',
+ *   alg: 'RS256',
+ *   use: 'sig',
+ *   ... // other parameters
+ * );
+ * ```
+ * @property kty Represents the key type.
+ * @property use Represents the intended use of the public key.
+ * @property alg Identifies the algorithm intended for use with the key.
+ * @property kid Key ID, unique identifier for the key.
+ * @property crv Elliptic curve name for EC keys.
+ * @property d Private key component for EC or OKP keys.
+ * @property x X coordinate for EC keys, or the public key for OKP.
+ * @property y Y coordinate for EC keys.
+ *
+ */
+public class Jwk(
+  public val kty: String,
+  public val crv: String,
+  public val use: String?,
+  public val alg: String?,
+  public var kid: String?,
+  public val d: String? = null,
```

Comment:
is there a reason we have to explicitly set this to `null`?

Reply:
nope, not sure why i put it there. we use the builder pattern anyway and don't use this constructor